File: support/pg/src/main/java/com/coffee/gu/PaymentGateway.java
package com.coffee.gu;


public interface PaymentGateway {
    PaymentGatewayProvider provider();
    PGConfirmResult confirm(PaymentGatewayConfirm confirm);
    PGPayment getByOrderKey(String orderKey);
    PGCancelResult cancel(PaymentGatewayCancel cancel);

}

File: support/pg/src/main/kotlin/com/coffee/gu/PaymentGateway.kt
package com.coffee.gu

interface PaymentGateway {
    fun provider(): PaymentGatewayProvider
    fun confirm(confirm: PaymentGatewayConfirm): PGConfirmResult
    fun getByOrderKey(orderKey: String): PGPayment
    fun cancel(cancel: PaymentGatewayCancel): PGCancelResult
}
